cadquery/cq.py

```diff
@@ -4230,6 +4229,60 @@ def offset2D(
 
         return self.newObject(rv)
 
+    def _locs(self: T) -> Iterable[Location]:
+        """
+        Convert items on the stack into locations.
+
+        """
+
+        plane = self.plane
+        obj = self.val() if self.objects else None
+        locs: Iterable[Location]
+
+        if isinstance(obj, (Vector, Shape)):
+            locs = (
+                Location(plane, obj.Center())
+                for obj in self.objects
+                if isinstance(obj, (Vector, Shape))
+            )
+        elif isinstance(obj, Location):
+            locs = (obj for obj in self.objects if isinstance(obj, Location))
+        else:
+            locs = (self.plane.location,)
```

Comment:
How come the behaviour here is different to `Workplane.eachpoint`? My first thought was to reuse `self._locs()` within `Workplane.eachpoint` but I can see they're not quite doing the same thing.

Reply:
It has a different purpose: `_locs()` is supposed to return absolute locations. NB: I simplified it in the last commit.